runtime/src/accounts_db.rs

```diff
@@ -9950,6 +10000,31 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn test_combine_multiple_slots_into_one_at_startup() {
+        solana_logger::setup();
+        let (db, slot1) = create_db_with_storages_and_index(false, 2, None);
+        let slot2 = slot1 + 1;
+
+        let initial_accounts = get_all_accounts(&db, slot1..(slot2 + 1));
+
+        let tf = TempDir::new().unwrap();
+        let stores = db
+            .storage
+            .all_slots()
+            .into_iter()
+            .map(|slot| {
+                let storage = db.storage.get_slot_storage_entry(slot).unwrap();
+                (storage.append_vec_id(), storage)
+            })
+            .collect::<HashMap<_, _>>();
+        AccountsDb::combine_multiple_slots_into_one_at_startup(tf.path(), 1000, slot1, &stores);
+
+        //db.storage.remove(&slot2, false);
+
+        compare_all_accounts(&initial_accounts, &get_all_accounts(&db, slot1..slot2));
```

Comment:
Is it intentional that when you are getting the accounts, `slot2` is included , while here it is not?
`slot1..(slot2+1)` in the line below vs `slot1..slot2` here:
```rust
    let initial_accounts = get_all_accounts(&db, slot1..(slot2 + 1));
```

Reply:
yes, but that code is now changed anyway. the fn I'm testing doesn't actually update the 'storages' dashmap.